File: src/dataset/typescript/minesweeper_exercism-samajammin/openAI/ChatGPT4_minesweeper_v1.ts
export default class Minesweeper {
  annotate(board: string[]): string[] {
    const mineCell = '*';
    const rows = board.length;
    const cols = board[0]?.length || 0;

    return board.map((row, rowIdx) => {
      const resultRow = Array(cols);
      for (let colIdx = 0; colIdx < cols; colIdx++) {
        if (row[colIdx] === mineCell) {
          resultRow[colIdx] = mineCell;
          continue;
        }
        let cellValue = 0;
        for (let r = Math.max(0, rowIdx - 1); r <= Math.min(rows - 1, rowIdx + 1); r++) {
          for (let c = Math.max(0, colIdx - 1); c <= Math.min(cols - 1, colIdx + 1); c++) {
            if (board[r][c] === mineCell) {
              cellValue++;
            }
          }
        }
        resultRow[colIdx] = cellValue === 0 ? ' ' : cellValue.toString();
      }
      return resultRow.join('');
    });
  }
}

File: src/dataset/python/minesweeper_exercism-fruit-in/openAI/ChatGPT4_minesweeper_v1.py
def annotate(minefield):
    if not minefield or any(len(row) != len(minefield[0]) for row in minefield):
        raise ValueError(r".+")

    rows, cols = len(minefield), len(minefield[0])
    result = []

    for row in range(rows):
        new_row = []
        for col in range(cols):
            cell = minefield[row][col]
            if cell == ' ':
                counter = sum(
                    1
                    for r in range(max(row - 1, 0), min(row + 2, rows))
                    for c in range(max(col - 1, 0), min(col + 2, cols))
                    if minefield[r][c] == '*'
                )
                new_row.append(str(counter) if counter > 0 else ' ')
            elif cell == '*':
                new_row.append('*')
            else:
                raise ValueError(r".+")
        result.append(''.join(new_row))

    return result